packages/dma-contracts/contracts/actions/common/PullTokenMaxAmount.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.15;
+
+import { Executable } from "../common/Executable.sol";
+import { SafeERC20, IERC20 } from "../../libs/SafeERC20.sol";
+import { PullTokenData } from "../../core/types/Common.sol";
+import "../../core/types/Common.sol";
+
+/**
+ * @title PullToken Action contract
+ * @notice Pulls token from a target address to the current calling context
+ */
+contract PullTokenMaxAmount is Executable {
```

Comment:
instead of this new action the `PullToken` and `TokenBalance` could be used.

modify `PullToken` to read from storage
`TokenBalance` writes the exact balance ( could also write token address and `from` address )
`PullToken` reads the amount from storage